### PR TITLE
87 js fundamentals

### PR DESCRIPTION
**Pull request by niksseif**
# Curriculum PR Template

### Does this PR close any issues?
*This PR closes [#87](https://github.com/turingschool/curriculum-site/issues/87)*

### Description
This PR addresses the following updates as requested in the ticket:

- [x] Updated verbiage from C# to Ruby
- [x] 	Fixed typos throughout the documentation
- [x] 	Improved overall style for better readability

### Additions after Nik's work
Upon realizing we were missing some key JS Fundamentals throughout the SE2 curriculum, I added those to this PR.  It makes for a big lesson but its all things they've seen and done in Ruby.  

Added:
- arrays and objects
- interpolation
- direct statement of key points
- for loops
- a repl for more hands on practice throughout 

Requesting a full review

### CODE REVIEW

**Comment by hfaerber** [comment]
We realized a few things were missing.  Objects, arrays, for loops, hands on practice and key points.  Those have been added now.  Requesting full review please.

File: module2/lessons/js_fundamentals.md [hunk 1]
```diff
@@ -29,85 +29,89 @@ Now that we know a little bit about JavaScript, let's talk about how to actually
...
+* **Null**
+  * an intentionally declared value of "nothing"
+  * maybe a placeholder value until data comes through to replace it
 
-As we're writing code, there are many times when we want to be able to store a value so we have ability to reuse it in multiple places. Writing the same thing over and over is a pain, and it's common that we need to have access to the same information more than once. A variable lets us do just that!.
+* **Number**
+  * any number - a negative, decimal or whole number (*not* in quotation marks)
+ 
+* **String**
+  * text data - any characters wrapped in quotation marks (including lettesr, numbers, symbols, spaces)
```
Comment (niksseif) [inline module2/lessons/js_fundamentals.md:84]: Thanks for adding this section here, just a small typo in here "lettesr"

File: module2/lessons/js_fundamentals.md [hunk 2]
```diff
@@ -133,6 +136,130 @@ let creatureCount = "<p>I have " + quantity + " very fancy" + mythicalCreature +
...
+
+Interpolation:  
+  - Uses the backticks and `${ }` to inject variables to make a big, long string.  
+
+```js
+var name = "Hillary";
+var age = 30;
+var location = "Denver";
+
+var interpolatedString = `My friend ${name} is ${age} years old and lives in ${Denver}.`
```
Comment (niksseif) [inline module2/lessons/js_fundamentals.md:241]: I am not sure if this is intentional, but inside the interpolation, the location variable has not been used, and “Denver” has been hardcoded.

**Review by niksseif — COMMENTED** [review]
@hfaerber Looks good, I left 2 minor suggestion. Thank you so much for working through this 🙏🏼

File: module2/lessons/js_fundamentals.md (hunk 2 above)
Comment (hfaerber) [reply module2/lessons/js_fundamentals.md:241]: _Thank you!_

**Review by hfaerber — COMMENTED** [review]
merged

**Pull request merged**